activestorage/app/controllers/concerns/active_storage/streaming.rb

```diff
@@ -61,6 +61,13 @@ def send_blob_stream(blob, disposition: nil) # :doc:
         blob.download do |chunk|
           stream.write chunk
         end
+      rescue ActiveStorage::FileNotFoundError
+        expires_now
+        head :not_found
+      rescue
+        expires_now
+        head :internal_server_error
```

Comment:
Do we still need to do this if we are raising below?

Reply:
The two lines are still needed. If you remove them and run the tests, they will fail. 
It is counter intuitive. Perhaps I should add a comment.
 
When downloading the file from the backing service for streaming, the 200 response + cache header are already set on the response object, they're just not yet committed to the client, so they can still be changed. 

When we raise that late (send_stream's block), we still send a 200 + caching header. 
I don't think a 500 will be common, but could still happen because of bad network or something similar.

Reply:
I added a comment.